.github/workflows/playground.yml

```diff
@@ -0,0 +1,98 @@
+name: WordPress Playground PR Testing
+
+# When to run tests.
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+jobs:
+  build-and-deploy:
+    # Name.
+    name: WordPress Playground
+
+    # Virtual Environment to use.
+    # @see: https://github.com/actions/virtual-environments
+    runs-on: ubuntu-latest
+
+    # Define permissions for this action.
+    permissions:
+      contents: write
+      pull-requests: write
+
+    # Environment Variables.
+    # Accessible by using ${{ env.NAME }}
+    # Use ${{ secrets.NAME }} to include any GitHub Secrets in ${{ env.NAME }}
+    # The base folder will always be /home/runner/work/github-repo-name/github-repo-name
+    env:
+      PLUGIN_SLUG: "convertkit" # The plugin's slug
+      PLAYGROUND_BRANCH_NAME: "pr-playground-${{ github.event.pull_request.number }}" # The branch to create from this PR which will have the built Plugin ZIP and blueprint.json added
+ 
+    # Steps to build and provide the Playground URL
+    steps:
+      # Checkout (copy) this repository's Plugin to this VM.
+      - name: Checkout Plugin
+        uses: actions/checkout@v4
+
+      # Installs Kit WordPress Libraries.
+      - name: Run Composer
+        run: composer install --no-dev
+
+      # Installs required packages that must be included in the Plugin
+      # as specified in composer.json's "require" section.
+      - name: Create ZIP File and Blueprint JSON
+        run: |
+          # Create ZIP file
+          zip -r ${{ env.PLUGIN_SLUG }}.zip . -x ".git/*" ".github/*" ".scripts/*" ".wordpress-org/*" "log/*" "tests/*" "*.md" "*.yml" "*.json" "*.neon" "*.lock" "*.xml" "*.dist" "*.example"
+
+          # Create blueprint.json
+          cat > blueprint.json << EOF
+          {
+            "landingPage": "/wp-admin/index.php",
+            "login": true,
+            "steps": [
+              {
+                "step": "installPlugin",
+                "pluginData": {
+                  "resource": "url",
+                  "url": "https://raw.githubusercontent.com/${{ github.repository }}/${{ env.PLAYGROUND_BRANCH_NAME }}/${{ env.PLUGIN_SLUG }}.zip"
+                }
+              }
+            ]
+          }
+          EOF
+
+      # Create Temporary Branch
```

Comment:
I'd prefer to upload the blueprint and plugin build ZIP somewhere else (S3?), instead of cluttering the repo with branches that hold the build files.

The location of the blueprint and ZIP file need to be public, so playground can read the blueprint and fetch and install the Plugin ZIP file.

Reply:
@ivanmp91 is there an available bucket we can use for these zips?

Reply:
Also in order to upload the new builds on this s3 bucket, you'll need to add one step in your Github Workflow that looks like this:

```
    - name: Configure AWS credentials
      uses: aws-actions/configure-aws-credentials@v3
      with:
        role-to-assume: arn:aws:iam::048876701201:role/KitWPPluginBuildsRole
        role-session-name: kit-wordpress
        aws-region: us-east-2
```

Then you should be able to use something like aws cli to upload those files to that s3 bucket.

Reply:
Sorry for the delay. Thanks for providing this @ivanmp91 - worked brilliantly.